### PR TITLE
[All hosts] (AppSource) Update Edge Legacy validation guidance

### PR DESCRIPTION
**Pull request by samantharamon**
AppSource no longer rejects add-in submissions that don't support Microsoft Edge Legacy (WebView1, EdgeHTML-based); it now only provides a warning to the submitter.

### CODE REVIEW

File: docs/concepts/browsers-used-by-office-web-add-ins.md [hunk 1]
```diff
@@ -58,13 +58,13 @@ To determine whether Office 2016 or Office 2019 is retail or volume-licensed, us
 | Office 2013 | Doesn't matter | Doesn't matter | Internet Explorer 11 |
 | Office 2016 | Volume-licensed | Doesn't matter | Internet Explorer 11 |
 | Office 2019 | Volume-licensed | Doesn't matter | Internet Explorer 11 |
-| Office 2016 to Office 2019 | Retail | No | Microsoft Edge<sup>1, 2</sup> with original WebView (EdgeHTML)</br>If Edge is not installed, Internet Explorer 11 is used. |
+| Office 2016 to Office 2019 | Retail | No | Microsoft Edge Legacy (EdgeHTML)<sup>1, 2</sup></br>If Edge isn't installed, Internet Explorer 11 is used. |
```
Comment (AlexJerabek) [inline docs/concepts/browsers-used-by-office-web-add-ins.md:61]: You already caveat (EdgeHTML) earlier in the article. Do you need to do it again here, or would "Microsoft Edge Legacy" be sufficient? If you do change it here, please change it later in the article as well.

File: docs/concepts/browsers-used-by-office-web-add-ins.md [hunk 2]
```diff
@@ -77,36 +77,36 @@ For subscription Office on Windows, the browser that is used is determined by th
...
+Also, Internet Explorer 11 doesn't support some HTML5 features such as media, recording, and location. To learn more, see [Determine at runtime if the add-in is running in IE11 or Microsoft Edge Legacy](../develop/support-ie-11.md#determine-at-runtime-if-the-add-in-is-running-in-ie11-or-microsoft-edge-legacy).
 
 ## Troubleshoot Microsoft Edge issues
 
-### Service Workers are not working
+### Service Workers aren't working
 
-Office Add-ins do not support Service Workers when the original Microsoft Edge WebView, [EdgeHTML](https://en.wikipedia.org/wiki/EdgeHTML), is used. They are supported with the [Chromium-based Edge WebView2](/microsoft-edge/hosting/webview2).
+Office Add-ins don't support Service Workers when Microsoft Edge Legacy's WebView, [EdgeHTML](https://en.wikipedia.org/wiki/EdgeHTML), is used. They're supported with the [Chromium-based Edge WebView2](/microsoft-edge/hosting/webview2).
```
Comment (AlexJerabek) [inline docs/concepts/browsers-used-by-office-web-add-ins.md:107]: Possessives on brand names sometimes don't translate well, so it's best to avoid them when possible.
```suggestion
Office Add-ins don't support Service Workers when the Microsoft Edge Legacy WebView, [EdgeHTML](https://en.wikipedia.org/wiki/EdgeHTML), is used. They're supported with the [Chromium-based Edge WebView2](/microsoft-edge/hosting/webview2).
```

File: docs/develop/support-ie-11.md [hunk 3]
```diff
@@ -1,27 +1,31 @@
 ---
-title: Support Internet Explorer 11
-description: Learn how to support Internet Explorer 11 and ES5 Javascript in your add-in.
-ms.date: 05/01/2022
+title: Support Internet Explorer 11 and Microsoft Edge Legacy
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:2]: I think this is fine as is, but did you consider titling this page "Support older Microsoft browsers and Office versions"?

File: docs/develop/support-ie-11.md [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
+ms.date: 12/12/2022
 ms.localizationpriority: medium
 ---
 
-# Support Internet Explorer 11
+# Support Internet Explorer 11 and Microsoft Edge Legacy
 
 > [!IMPORTANT]
-> **Internet Explorer still used in Office Add-ins**
+> **Internet Explorer and Microsoft Edge Legacy still used in Office Add-ins**
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:11]: I know this is acting like a header, but I think the verb helps readability.
```suggestion
> **Internet Explorer and Microsoft Edge Legacy are still used in Office Add-ins**
```

File: docs/develop/support-ie-11.md [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
 >
-> Some combinations of platforms and Office versions, including perpetual versions through Office 2019, still use the webview control that comes with Internet Explorer 11 to host add-ins, as explained in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). We recommend (but don't require) that you continue to support these combinations, at least in a minimal way, by providing users of your add-in a graceful failure message when your add-in is launched in the Internet Explorer webview. Keep these additional points in mind:
+> Some combinations of platforms and Office versions, including perpetual versions through Office 2019, still use the webview controls that come with Internet Explorer 11 and Microsoft Edge Legacy (EdgeHTML-based) to host add-ins, as explained in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). We recommend (but don't require) that you continue to support these combinations, at least in a minimal way, by providing users of your add-in a graceful failure message when your add-in is launched in these webviews. Keep these additional points in mind:
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:13]: Brevity suggestion.
```suggestion
> Some combinations of platforms and Office versions, including perpetual versions through Office 2019, still use the webview controls that come with Internet Explorer 11 and Microsoft Edge Legacy (EdgeHTML-based) to host add-ins, as explained in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). We recommend (but don't require) that you support these combinations, at least in a minimal way, by providing users of your add-in a graceful failure message when your add-in is launched in these webviews. Keep these additional points in mind:
```

File: docs/develop/support-ie-11.md [hunk 6]
```diff
@@ -1,27 +1,31 @@
...
 >
-> - Office on the web no longer opens in Internet Explorer. Consequently, [AppSource](/office/dev/store/submit-to-appsource-via-partner-center) no longer tests add-ins in Office on the web using Internet Explorer as the browser.
-> - AppSource still tests for combinations of platform and Office *desktop* versions that use Internet Explorer, however it only issues a warning when the add-in does not support Internet Explorer; the add-in is not rejected by AppSource.
+> - [AppSource](/office/dev/store/submit-to-appsource-via-partner-center) no longer tests add-ins in Office on the web using Internet Explorer or Microsoft Edge Legacy. Office on the web no longer opens in Internet Explorer.
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:15]: Can someone have Edge Legacy as a browser anymore? These articles are written from a browser-centric perspective, even though it's really the Office version that's dictating the webview.

File: docs/develop/support-ie-11.md [hunk 7]
```diff
@@ -1,27 +1,31 @@
...
+> - AppSource still tests for combinations of platform and Office *desktop* versions that use Internet Explorer and Microsoft Edge Legacy. However, it only issues a warning when the add-in doesn't support these browsers; the add-in isn't rejected by AppSource.
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:16]: Since this is two sentences, I'd favor the period over the semicolon. Feel free to disagree.
```suggestion
> - AppSource still tests for combinations of platform and Office *desktop* versions that use Internet Explorer and Microsoft Edge Legacy. However, it only issues a warning when the add-in doesn't support these browsers. The add-in isn't rejected by AppSource.
```

File: docs/develop/support-ie-11.md [hunk 8]
```diff
@@ -1,27 +1,31 @@
...
 
-If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11). For information about which combinations of Windows and Office use the IE11-based browser control, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
+If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11) or Microsoft Edge Legacy (EdgeHTML-based). For information about which combinations of Windows and Office use these browser controls, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
+
+For more information on how to support each browser, select the applicable tab.
+
+# [Internet Explorer](#tab/ie)
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:25]: Why tabs instead of H2 sections?

File: docs/develop/support-ie-11.md [hunk 9]
```diff
@@ -1,27 +1,31 @@
...
 > - The [Script Lab tool](../overview/explore-with-script-lab.md) no longer supports Internet Explorer.
 
 Office Add-ins are web applications that are displayed inside IFrames when running on Office on the web. Office Add-ins are displayed using embedded browser controls when running in Office on Windows or Office on the Mac. The embedded browser controls are supplied by the operating system or by a browser installed on the user's computer.
 
-If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11). For information about which combinations of Windows and Office use the IE11-based browser control, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
+If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11) or Microsoft Edge Legacy (EdgeHTML-based). For information about which combinations of Windows and Office use these browser controls, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:21]: I think you've defined the"Edge-based" bit earlier in the article.
```suggestion
If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11) or Microsoft Edge Legacy. For information about which combinations of Windows and Office use these browser controls, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
```

File: docs/develop/support-ie-11.md [hunk 10]
```diff
@@ -39,50 +43,76 @@ There are two popular transpilers. Both of them can work with source files that
...
-    // IE is the browser. So here, do one of the following: 
-    //  1. Provide an alternate experience that does not use any of the HTML5
-    //     features that are not supported in IE.
-    //  2. Enable the add-in to gracefully fail by putting a message in the UI that
-    //     says something similar to: 
-    //      "This add-in won't run in your version of Office. Please upgrade 
-    //      either to perpetual Office 2021 or to a Microsoft 365 account."          
+---
+
+## Determine at runtime if the add-in is running in IE11 or Microsoft Edge Legacy
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:77]: Should this section go earlier in the article?

File: docs/develop/support-ie-11.md [hunk 11]
```diff
@@ -39,50 +43,76 @@ There are two popular transpilers. Both of them can work with source files that
...
+
+Your add-in can discover if it's running in IE11 or Microsoft Edge Legacy by reading the [window.navigator.userAgent](https://developer.mozilla.org/docs/Web/API/Navigator/userAgent) property. This enables the add-in to either provide an alternate experience or gracefully fail. The following is an example.
 
+```javascript
+if (navigator.userAgent.indexOf("Trident") !== -1) {
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:82]: Shouldn't this be `===`?

File: docs/develop/support-ie-11.md [hunk 12]
```diff
@@ -39,50 +43,76 @@ There are two popular transpilers. Both of them can work with source files that
...
+    /*
+       IE11 is the browser in use. Do one of the following:
+        1. Provide an alternate add-in experience that doesn't use any of the HTML5
+           features that aren't supported in IE11.
+        2. Enable the add-in to gracefully fail by adding a message to the UI that
+           says something similar to:
+           "This add-in won't run in your version of Office. Please upgrade either to
+           perpetual Office 2021 or to a Microsoft 365 account."
+    */
+} else if (navigator.userAgent.indexOf("Edge") !== -1) {
```
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:92]: Same equal question?

File: docs/testing/test-debug-office-add-ins.md [hunk 13]
```diff
@@ -17,9 +17,17 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
 > On Windows computers, the version of Windows and Office will determine which browser control is used by add-ins. For more information, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
 
 > [!IMPORTANT]
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:19]: This IMPORTANT note is getting huge. Should it become an H2 section?

File: docs/testing/test-debug-office-add-ins.md [hunk 14]
```diff
@@ -17,9 +17,17 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
...
+>
+> AppSource doesn't use Internet Explorer or Microsoft Edge Legacy (EdgeHTML) to test add-ins in Office on the web. However, if a significant number of your users will use Edge Legacy to open Office on the web, then you should test with it. (Office on the web will not open in Internet Explorer, so you can't, and don't need to, test Office on the web with Internet Explorer.) For more information, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues). Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues/new/choose) repo.
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:26]: Same question here as earlier about the availability of old Edge as a browser.

Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:26]: I think we should link to the issues list, rather than the new issue template. That way, the user can check for duplicates.
```suggestion
> AppSource doesn't use Internet Explorer or Microsoft Edge Legacy (EdgeHTML) to test add-ins in Office on the web. However, if a significant number of your users will use Edge Legacy to open Office on the web, then you should test with it. (Office on the web will not open in Internet Explorer, so you can't, and don't need to, test Office on the web with Internet Explorer.) For more information, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues). Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues) repo.
```

File: docs/testing/test-debug-office-add-ins.md [hunk 15]
```diff
@@ -17,9 +17,17 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
...
+>
+> ***Add-ins tested for Office on Windows***
+>
+> Because some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy, AppSource tests whether your add-in supports these browsers. If your add-in doesn't support these browsers, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md).
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:30]: Consider breaking up this first sentence to make it easier to parse.
```suggestion
> Some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy. AppSource tests whether your add-in supports these browsers. If your add-in doesn't support these browsers, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md).
```

**Review by AlexJerabek — COMMENTED** [review]
Thanks for putting this together @samantharamon. I have a few questions before signing off, but looks good overall.

File: docs/concepts/browsers-used-by-office-web-add-ins.md (hunk 1 above)
Comment (samantharamon) [reply docs/concepts/browsers-used-by-office-web-add-ins.md:61]: The EdgeHTML reference first appears in the Important block. My thought is that if the user is directly linked to the "Perpetual versions" section or somehow skips the Important block, they would have another way to differentiate between the old and new Edge versions through that EdgeHTML reference.

File: docs/develop/support-ie-11.md (hunk 3 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:2]: That's a great suggestion and would definitely help with updating this article in the future. I'll make the change.

File: docs/develop/support-ie-11.md (hunk 6 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:15]: Good point. Even if Edge Legacy were somehow still installed on older Windows OS versions, such as Windows 7 and 8/8.1, users would be prompted to update to the Chromium-based version to be able to use the browser. I'll update line 15 (and the other article where this is referenced) to include Edge Legacy.

File: docs/develop/support-ie-11.md (hunk 9 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:21]: Similar to my earlier comment about this being the first EdgeHTML reference outside the Important note block. Although, I'm open to changing this if we think the note block mention is sufficient.

File: docs/develop/support-ie-11.md (hunk 8 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:25]: I figured it would be easier to reach the Edge Legacy sections (if that's what the user is after) without having to scroll through the IE sections.

File: docs/develop/support-ie-11.md (hunk 10 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:77]: That makes sense. I'll move it up.

File: docs/develop/support-ie-11.md (hunk 11 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:82]: No. A -1 value indicates that the specified browser isn't in use. Because we can't guarantee the index of the browser in the returned string, it's safer to do this check instead of using a specific value.

File: docs/develop/support-ie-11.md (hunk 12 above)
Comment (samantharamon) [reply docs/develop/support-ie-11.md:92]: See earlier comment.

File: docs/testing/test-debug-office-add-ins.md (hunk 13 above)
Comment (samantharamon) [reply docs/testing/test-debug-office-add-ins.md:19]: I was going back and forth on this. Now that I look at it again, the note does take up majority of the article space now. I'll move this into its own section.

File: docs/develop/support-ie-11.md (hunk 11 above)
Comment (AlexJerabek) [reply docs/develop/support-ie-11.md:82]: Ah, gotcha. I had to re-read that to make sense of the double negative.

File: docs/testing/test-debug-office-add-ins.md [hunk 16]
```diff
@@ -16,18 +16,19 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
...
->
-> ***Add-ins tested for Office on Windows***
->
-> Because some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy, AppSource tests whether your add-in supports these browsers. If your add-in doesn't support these browsers, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md).
+## Test for AppSource validation
+
+Add-ins marketed through AppSource go through a validation process that includes testing on all platforms.
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:21]: This is probably out of scope for this PR, but this segment here now feels redundant. The prior section talks about cross-plat. Maybe it's just me.

File: docs/testing/test-debug-office-add-ins.md [hunk 17]
```diff
@@ -16,18 +16,19 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
...
+Add-ins are tested for Office on the web with all major modern browsers, including Microsoft Edge (Chromium-based WebView2), Chrome, and Safari. Accordingly, you should test on these platforms and browsers before you submit to AppSource. For more information about validation, see [Commercial marketplace certification policies](/legal/marketplace/certification-policies), especially [section 1120.3](/legal/marketplace/certification-policies#11203-functionality), and the [Office Add-in application and availability page](/javascript/api/requirement-sets).
+
+Office on the web no longer opens in Internet Explorer or Microsoft Edge Legacy (EdgeHTML). Consequently, AppSource doesn't test Office on the web on these browsers. For more information, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues). Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues) repository.
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:27]: Consider putting the "learn more" link at the end of the paragraph.
```suggestion
Office on the web no longer opens in Internet Explorer or Microsoft Edge Legacy (EdgeHTML). Consequently, AppSource doesn't test Office on the web on these browsers. Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues) repository. For more information, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues).
```

File: docs/testing/test-debug-office-add-ins.md [hunk 18]
```diff
@@ -16,18 +16,19 @@ Office Add-ins run across major platforms, so you need to test an add-in in all
...
+
+### Add-ins tested for Office on Windows
+
+Some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy. AppSource tests whether your add-in supports these browsers. If your add-in doesn't support these browsers, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md).
```
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:31]: Some technicality phrasing.
```suggestion
Some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy. AppSource tests whether your add-in supports these browser controls. If your add-in doesn't support these browser controls, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md).
```

**Review by AlexJerabek — APPROVED** [review]
Approving with a couple more suggestions. Whether you take or leave them, I'm good with this being checked in without giving another review.
